src/lib/autoinstall/script.rb

```diff
@@ -268,20 +268,25 @@ def initialize(hash)
       "perl"   => "/usr/bin/perl",
       "python" => "/usr/bin/python"
     }.freeze
+
     # Runs the script
+    #
+    # @param env [Hash] hash representing a set of environment variables
     # @return [Boolean,nil] if exit code is zero; nil if the script was not executed
-    def execute
+    def execute(env = {})
       return if already_run? && !rerun
 
       # TODO: maybe own class for interpreters?
       cmd = INTERPRETER_MAP[interpreter] || interpreter
       debug_flag = debug ? (DEBUG_FLAG_MAP[interpreter] || "") : ""
       params_s = params.join(" ") # shell escaping is up to user, see documentation
 
+      cmd_line = "#{cmd} #{debug_flag} #{script_path.shellescape} #{params_s} " \
+          "&> #{log_path.shellescape}"
+      cmd_line = "#{env_vars(env)} #{cmd_line}" unless env.empty?
```

Comment:
I think condition is not needed here as shell ignores white spaces before binary. It will be consistent with debug flag.